src/vs/code/electron-main/app.ts

```diff
@@ -609,6 +610,16 @@ export class CodeApplication extends Disposable {
 		// Post Open Windows Tasks
 		this.afterWindowOpen();
 
+		// macOS: detect non-standard app installation
+		if (isMacintosh && !app.isInApplicationsFolder()) {
```

Comment:
Should this move at the end of the `afterWindowOpen` method?

Reply:
Kept it here for the servicesaccessor, do you want to expose it in `afterWindowOpen`

Reply:
I was just wondering why it would be after `afterWindowOpen` and not inside that method, because logically it is after window open, so why not have it in there.